File: src/endfield/entities/pattern/ShootBursts.java
package endfield.entities.pattern;

import mindustry.entities.pattern.ShootPattern;
import org.jetbrains.annotations.Nullable;

public class ShootBursts extends ShootPattern {
	public int shotsPerBurst = 1;
	public float spread = 5f;
	public int bursts = 3;
	public float burstDelay = 30f;

	public ShootBursts(int shot, int bur, float spr) {
		shotsPerBurst = shot;
		bursts = bur;
		spread = spr;

		// This is so that stats work correctly
		shots = bursts * shotsPerBurst;
	}

	public ShootBursts() {}

	@Override
	public void shoot(int totalShots, BulletHandler handler, @Nullable Runnable barrelIncrementer) {
		for (int i = 0; i < bursts; i++) {
			for (int j = 0; j < shotsPerBurst; j++) {
				float angleOffset = j * spread - (shotsPerBurst - 1) * spread / 2f;
				handler.shoot(0, 0, angleOffset, firstShotDelay + shotDelay * j + burstDelay * i);
			}
		}
	}
}

File: src/heavyindustry/entities/pattern/ShootBursts.kt
package heavyindustry.entities.pattern

import arc.util.Nullable
import mindustry.entities.pattern.ShootPattern

open class ShootBursts : ShootPattern {
	@JvmField var shotsPerBurst = 1
	@JvmField var spread = 5f
	@JvmField var bursts = 3
	@JvmField var burstDelay = 30f

	constructor(shot: Int, bur: Int, spr: Float) {
		shotsPerBurst = shot
		bursts = bur
		spread = spr

		// This is so that stats work correctly
		shots = bur * shot
	}

	constructor()

	override fun shoot(totalShots: Int, handler: BulletHandler, @Nullable barrelIncrementer: Runnable) {
		for (i in 0..<bursts) {
			for (j in 0..<shotsPerBurst) {
				val angleOffset = j * spread - (shotsPerBurst - 1) * spread / 2f
				handler.shoot(0f, 0f, angleOffset, firstShotDelay + shotDelay * j + burstDelay * i)
			}
		}
	}
}
